src/components/text/List/List.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useMemo } from 'react'
+import { View } from 'react-native'
+import { Box } from '../../structure'
+import { BodyText } from '../BodyText/BodyText'
+import { useStyles } from '../../../theme'
+import { Item, ItemProps } from './Item/Item'
+
+type ListType = 'bullet' | 'numbered'
+
+export interface ListWithDataSourceProps {
+  /**
+   * Strings to display in the List.
+   */
+  dataSource: string[]
+}
+
+export interface ListWithChildrenProps {
+  /**
+   * Strings to display in the list (can be stylized using `TextStyle` component).
+   */
+  children: React.ReactNode
+}
+
+export type ListProps = {
+  /**
+   * Type of list displayed.
+   */
+  type?: ListType
+  /**
+   * Strings to display in the List.
+   */
+  dataSource?: string[]
+  /**
+   * Strings to display in the list (can be stylized using `TextStyle` component).
+   */
+  children?: React.ReactNode
+} & (ListWithDataSourceProps | ListWithChildrenProps)
+
+/**
+ * Component displaying texts in a List.
+ */
+export const List: React.FC<ListProps> & { Item: typeof Item } = ({
+  type = 'bullet',
+  dataSource,
+  children,
+}) => {
+  const items = useItems(dataSource, children, type)
+
+  return <Box space="small">{items}</Box>
+}
+
+/**
+ * Get items to display in the list
+ */
+const useItems = (
+  dataSource: string[] | undefined,
+  children: React.ReactNode | undefined,
+  type: ListType,
+): React.ReactNode =>
+  useMemo(() => {
+    if (dataSource && children) {
+      console.warn(
+        'Both children and dataSource props are defined in List component. Only dataSource will be considered.',
+      )
+    }
+
+    let items = children
+
+    if (dataSource) {
+      items = dataSource.map((item, i) => <Item key={i}>{item}</Item>)
+    }
+
+    return React.Children.map(items, (item, index) => (
+      <Box horizontal space="small">
+        {_getBullet(type, index)}
+        {item}
+      </Box>
+    ))
```

Comment:
Yes, however I think it is more readable as it is right now, as moving the `React.children.map` out of the hook would require having another hook to prevent the mapping function to be called every single time.

E.g.
```tsx
const items = useItems(values, children, type)

const childrenReal = useMemo(() => React.Children.map(items, (item, index) => (
    <Box horizontal space="small">
      {_getBullet(type, index)}
      {item}
    </Box>
)), [items, type])
return <Box space="small">{childrenReal}</Box>
```

Both approaches are acceptable though.

Reply:
I have a preference for looping in `List`, but go with what you find is the best readability vs. performance tradeoff 👍

Reply:
Changed the code as you suggested, it improved readability a bit and should have almost none performance impact. :wink: